src/main/java/io/vertx/core/cli/OptionModel.java

```diff
@@ -0,0 +1,371 @@
+   * short name.
+   */
+  public static final String NO_NAME = "\0";
+
+  /**
+   * the option long name.
+   */
+  protected String longName = NO_NAME;
+
+  /**
+   * the option short name.
+   */
+  protected String shortName = NO_NAME;
+
+  /**
+   * the option name used in usage message.
+   */
+  protected String argName = DEFAULT_ARG_NAME;
+
+  /**
+   * The option description.
+   */
+  protected String description;
+
+  /**
+   * whether or not the option is required. A mandatory not set throws a {@link MissingOptionException}.
+   */
+  protected boolean required;
+
+  /**
+   * whether or not the option is hidden. Hidden options are not displayed in usage.
+   */
+  protected boolean hidden;
+
+  /**
+   * whether or not the option receives a single value. {@code true} by default.
+   */
+  protected boolean singleValued = true;
+
+  /**
+   * whether or not the option can recevie multiple values.
+   */
+  protected boolean multiValued;
+
+  /**
+   * the option default value.
+   */
+  protected String defaultValue;
+
+  /**
+   * whether or not the option is a flag. Flag option does not require a value. If an option is a flag, it is
+   * evaluated to {@link true} if the option is used in the command line.
+   */
+  protected boolean flag;
+
+  /**
+   * Creates a new empty instance of {@link OptionModel}.
+   */
+  public OptionModel() {
+  }
+
+  /**
+   * Creates a new instance of {@link OptionModel} by copying the state of another {@link OptionModel}.
+   *
+   * @param other the other option
+   */
+  public OptionModel(OptionModel other) {
+    this();
+    this.longName = other.longName;
+    this.shortName = other.shortName;
+    this.argName = other.argName;
+    this.description = other.description;
+    this.required = other.required;
+    this.hidden = other.hidden;
+    this.singleValued = other.singleValued;
+    this.multiValued = other.multiValued;
+    this.defaultValue = other.defaultValue;
+    this.flag = other.flag;
+  }
+
+  /**
+   * Creates a new instance of {@link OptionModel} from the given {@link JsonObject}
+   *
+   * @param json the json object representing the option
+   * @see #toJson()
+   */
+  public OptionModel(JsonObject json) {
+    this();
+    OptionModelConverter.fromJson(json, this);
+  }
+
+  /**
+   * Gets the json representation of this {@link OptionModel}.
+   *
+   * @return the json representation
+   */
+  public JsonObject toJson() {
+    JsonObject json = new JsonObject();
+    OptionModelConverter.toJson(this, json);
+    return json;
+  }
+
+  /**
+   * Checks whether or not the option is valid. This implementation check that it has a short name or a long name.
+   * This method is intended to be extended by sub-class. {@link CommandLineParser} should check that the set of
+   * option of a {@link CLI} is valid before starting the parsing.
+   *
+   * @return whether or not the configuration of this option is valid.
+   */
+  public boolean isValid() {
```

Comment:
weird method return says "whether or not the configuration of this option is valid." however it always returns true or throw an exception. one of the two statements if false.